packages/zod-nestjs/src/lib/create-zod-dto.ts

```diff
@@ -76,12 +76,16 @@ export const createZodDto = <T extends OpenApiZodAny>(
          * array to a boolean.
          */
         const schemaObject = properties[key] as SchemaObjectForMetadataFactory;
-        const schemaObjectWithRequiredField = {
+        const schemaObjectWithFixedFields = {
           ...schemaObject,
+          type: schemaObject.type ? schemaObject.type[0] : schemaObject.type,
```

Comment:
We should remove `'null'` from type array then. Besides, this doesn't work with string values (rather than arrays) which TypeScript allows here.

Reply:
I think my changes make this work, but there's a case (or two) which I'm not sure about: `type: []` or `type: ["null"]`. What do you think should happen then?